templates/master.yaml.tmpl

```diff
@@ -774,13 +776,49 @@ systemd:
       RemainAfterExit=yes
       [Install]
       WantedBy=multi-user.target
+
+  {{ if eq .Provider "aws" -}}
+  - name: get-vault-token.service
+    enabled: true
+    contents: |
+      [Unit]
+      Description=get-vault-token
+      Requires=get-vault-ca.service k8s-setup-network-env.service docker.service wait-for-domains.service
+      After=get-vault-ca.service k8s-setup-network-env.service docker.service wait-for-domains.service
+
+      [Service]
+      EnvironmentFile=/etc/environment
+      EnvironmentFile=/etc/network-environment
+      Environment=VAULT_ADDR=https://{{ .VaultDomainName }}
+      Environment=CLUSTER_NAME={{ .ClusterName }}
+      Type=oneshot
+      RemainAfterExit=yes
+      ExecStartPre=/usr/bin/mkdir -p /etc/kubernetes/ssl/
+      ExecStartPre=/usr/bin/mkdir -p /etc/tokens
+      ExecStartPre=/bin/bash -c "while ! docker run --rm -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt {{.DockerRegistry}}/giantswarm/curl:7.67.0 -k -q --silent -o /dev/null https://{{ .VaultDomainName }};  do sleep 2s;echo wait for Vault;done;"
+      ExecStart=/bin/bash -c '\
+          export vault_token=$(\
+          docker run --rm -i\
+          -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt\
+          --net host\
+          --privileged=true\
+          -e VAULT_ADDR\
+          -e CLUSTER_NAME\
+          {{.DockerRegistry}}/giantswarm/vault:1.6.1\
+          login -method=aws role=$${CLUSTER_NAME}-master region={{ .AWSRegion }} -format=json | jq -r .auth.client_token);\
+          echo "VAULT_TOKEN=$${vault_token}" > /etc/tokens/node;\
```

Comment:
when the aws command fails the   main  `ExecStart` command exits with 0 (due to the last command  doing just echo) thus silently falining

Reply:
We can replace with `/bin/bash -Oe -c '\`